exchanges/kucoin/kucoin_wrapper.go

```diff
@@ -0,0 +1,1480 @@
+			if err != nil {
+				return nil, err
+			}
+			orders = append(orders, order.Detail{
+				OrderID:         spotOrders.Items[x].ID,
+				Amount:          spotOrders.Items[x].Size,
+				RemainingAmount: spotOrders.Items[x].Size - spotOrders.Items[x].DealSize,
+				ExecutedAmount:  spotOrders.Items[x].DealSize,
+				Exchange:        ku.Name,
+				Date:            spotOrders.Items[x].CreatedAt.Time(),
+				Price:           spotOrders.Items[x].Price,
+				Side:            side,
+				Type:            oType,
+				Pair:            dPair,
+			})
+		}
+	default:
+		return nil, fmt.Errorf("%w %v", asset.ErrNotSupported, getOrdersRequest.AssetType)
+	}
+	return orders, nil
+}
+
+// GetOrderHistory retrieves account order information
+// Can Limit response to specific order status
+func (ku *Kucoin) GetOrderHistory(ctx context.Context, getOrdersRequest *order.MultiOrderRequest) (order.FilteredOrders, error) {
+	if getOrdersRequest == nil {
+		return nil, common.ErrNilPointer
+	}
+	err := ku.CurrencyPairs.IsAssetEnabled(getOrdersRequest.AssetType)
+	if err != nil {
+		return nil, err
+	}
+	if getOrdersRequest.Validate() != nil {
+		return nil, err
+	}
+	var sideString string
+	sideString, err = ku.orderSideString(getOrdersRequest.Side)
+	if err != nil {
+		return nil, err
+	}
+	var orders []order.Detail
+	var orderSide order.Side
+	var orderStatus order.Status
+	var oType order.Type
+	var pair currency.Pair
+	switch getOrdersRequest.AssetType {
+	case asset.Futures:
+		var futuresOrders *FutureOrdersResponse
+		var newOrders *FutureOrdersResponse
+		if len(getOrdersRequest.Pairs) == 0 {
+			futuresOrders, err = ku.GetFuturesOrders(ctx, "", "", sideString, getOrdersRequest.Type.Lower(), getOrdersRequest.StartTime, getOrdersRequest.EndTime)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			for x := range getOrdersRequest.Pairs {
+				getOrdersRequest.Pairs[x], err = ku.FormatExchangeCurrency(getOrdersRequest.Pairs[x], getOrdersRequest.AssetType)
+				if err != nil {
+					return nil, err
+				}
+				newOrders, err = ku.GetFuturesOrders(ctx, "", getOrdersRequest.Pairs[x].String(), sideString, getOrdersRequest.Type.Lower(), getOrdersRequest.StartTime, getOrdersRequest.EndTime)
+				if err != nil {
+					return nil, fmt.Errorf("%w while fetching for symbol %s", err, getOrdersRequest.Pairs[x].String())
+				}
+				if futuresOrders == nil {
+					futuresOrders = newOrders
+				} else {
+					futuresOrders.Items = append(futuresOrders.Items, newOrders.Items...)
+				}
+			}
+		}
+		var enabledPairs currency.Pairs
+		enabledPairs, err = ku.GetEnabledPairs(asset.Futures)
+		if err != nil {
+			return nil, err
+		}
+		orders = make(order.FilteredOrders, len(futuresOrders.Items))
+		for i := range orders {
+			orderSide, err = order.StringToOrderSide(futuresOrders.Items[i].Side)
+			if err != nil {
+				return nil, err
+			}
+			pair, err = enabledPairs.DeriveFrom(futuresOrders.Items[i].Symbol)
+			if err != nil {
+				return nil, err
+			}
```

Comment:
```suggestion
if err !=nil {
    if errors.Is(err, currency.ErrPairNotFound) {
        continue
    }
    return nil, err
}
```

Just a bit nicer for the user